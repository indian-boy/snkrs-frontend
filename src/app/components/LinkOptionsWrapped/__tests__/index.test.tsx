import { render } from '@testing-library/react';
import * as React from 'react';
import { LinkOptionsWrapped } from '..';

jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: str => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
}));

const mockData = {
  title: 'Help',
  links: [
    {
      label: 'Home',
      url: '/',
    },
    {
      label: 'y',
      url: '/y',
    },
  ],
};

describe('<LinkOptionsWrapped  />', () => {
  it('should match snapshot', () => {
    const linkOptionsWrapped = render(<LinkOptionsWrapped {...mockData} />);
    expect(linkOptionsWrapped.container.firstChild).toMatchSnapshot();
  });
});
