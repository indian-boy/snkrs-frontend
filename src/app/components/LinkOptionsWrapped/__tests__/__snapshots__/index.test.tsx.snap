// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkOptionsWrapped  /> should match snapshot 1`] = `
.c0 {
  width: 100%;
  background: #000000;
  overflow: hidden;
  position: relative;
  color: #FFFFFF;
  padding-bottom: 1rem;
}

.c0[open] > summary .icon:after {
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.c1 {
  display: block;
  background: #000000;
  padding: 1rem 3rem 1rem 1rem;
  position: relative;
  cursor: pointer;
}

.c1:before {
  content: '';
  background: #FFFFFF;
  position: absolute;
  bottom: 0;
  height: 1px;
  width: calc(100% - 2.5rem);
  left: 0;
}

.c1::-webkit-details-marker {
  display: none;
}

.c2 {
  cursor: pointer;
}

.c2:before,
.c2:after {
  content: '';
  position: absolute;
  background: #FFFFFF;
  -webkit-transition: 0.2s all ease-in-out;
  transition: 0.2s all ease-in-out;
}

.c2:before {
  top: 50%;
  left: calc(100% - 2rem);
  right: 0.7rem;
  height: 0.1rem;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.c2:after {
  top: 1rem;
  right: 1.3rem;
  bottom: 0.9rem;
  width: 0.1rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1rem;
  margin: 1rem;
}

.c4 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #FFFFFF;
}

.c4:visited {
  color: #D6D6D6;
}

<details
  class="c0"
>
  <summary
    class="c1"
  >
    Help
    <span
      class="c2 icon"
    />
  </summary>
  <div
    class="c3"
  >
    <a
      aria-current="page"
      class="c4 active"
      href="/"
    >
      Home
    </a>
    <a
      class="c4"
      href="/y"
    >
      y
    </a>
  </div>
</details>
`;
